pkg/sql/logictest/testdata/logic_test/datetime

```diff
@@ -1523,7 +1523,7 @@ ORDER BY
 01:00:00               00:30:00          02:00:00                00:30:00          02:00:00                04:14:01.320914                 00:14:10.32
 1 day                  12:00:00          2 days                  12:00:00          2 days                  4 days 05:36:31.701948          05:40:07.68
 1 mon                  15 days           2 mons                  15 days           2 mons                  4 mons 7 days 00:15:51.058425   7 days 02:03:50.4
-1 mon 2 days 04:00:00  16 days 02:00:00  2 mons 4 days 08:00:00  16 days 02:00:00  2 mons 4 days 08:00:00  4 mons 15 days 28:24:59.745978  7 days 14:20:47.04
+1 mon 2 days 04:00:00  16 days 02:00:00  2 mons 4 days 08:00:00  16 days 02:00:00  2 mons 4 days 08:00:00  4 mons 15 days 28:24:59.745977  7 days 14:20:47.04
```

Comment:
hmm, this does not match postgresql.

in particular, psql returns: 
```
otan=# select '1 month 2 days 4 hours'::INTERVAL / .2362;
            ?column?
--------------------------------
 4 mons 15 days 28:24:59.778753
(1 row)
```

I'm not sure what's broken, but do you mind taking a look while you're here? Here is the PostgreSQL implementation of integer division: https://github.com/postgres/postgres/blob/e56bce5d43789cce95d099554ae9593ada92b3b7/src/backend/utils/adt/timestamp.c#L3266-L3304

Reply:
i feel like the answer is:

```
retNanos := time.Duration( int64(time.Duration(float64(d.nanos)).Microseconds()) / x) * time.Microsecond)
```

note the difference here -- we're dividing the microseconds by x, rather than the nanoseconds!